File: 055_JumpGame.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int size(nums.size());
		if(1 >= size)
			return true;
		int i = size - 2;
		while(i > 0){
			if(nums[i] + i >= size - 1)
				size = i + 1;
			i--;
		}
		return nums[0] >= size - 1;		
    }
};

File: 055_JumpGame.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        size = len(nums)
        i = size - 2
        for num in nums[1:-1][::-1]:
            if i + num >= size - 1:
                size = i + 1
            i -= 1
        return nums[0] >= size - 1